### ISSUE TITLE
Does pytorch-lightning suppot dynamic plot during training like https://github.com/stared/livelossplot in keras

### ISSUE THREAD

**Issue by wangbingnan136** [Lightning-AI/lightning#13174]
## 🚀 Feature

Does pytorch-lightning suppot dynamic plot during training like https://github.com/stared/livelossplot in keras
### Motivation

A dynamic plot of loss or metrics is quite useful for model training

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @awaelchli @edward-io @ananthsub @rohitgr7 @kamil-kaczmarek @Raalsky @Blaizzy

**Comment by awaelchli** [comment]
Have you tried tensorboard or any of the other [loggers](https://pytorch-lightning.readthedocs.io/en/stable/common/loggers.html)?

**Comment by nizioleque** [comment]
@wangbingnan136 I've created this Lightning logger for LiveLossPlot:

```python
from lightning.pytorch.loggers.logger import Logger
from lightning.pytorch.utilities import rank_zero_only
from livelossplot import PlotLosses


class LiveLossPlotLogger(Logger):
    def __init__(self):
        super().__init__()
        self.liveloss = PlotLosses()
        self.current_metrics = {}

    @property
    def name(self):
        return "LiveLossPlotLogger"

    @property
    def version(self):
        return "0.1"

    @rank_zero_only
    def log_hyperparams(self, params):
        pass

    @rank_zero_only
    def log_metrics(self, metrics, step):
        if "val_acc" in metrics:
            self.current_metrics["val_acc"] = metrics["val_acc"]
            self.current_metrics["val_loss"] = metrics["val_loss"]

        if "train_acc_epoch" in metrics:
            self.current_metrics["acc"] = metrics["train_acc_epoch"]
            self.current_metrics["loss"] = metrics["train_loss_epoch"]

            self.liveloss.update(self.current_metrics)
            self.liveloss.send()

            self.current_metrics = {}
```

Here is my usage example:
```python
    def training_step(self, batch, batch_idx):
        x, y = batch
        logits = self(x)

        loss = self.loss_fn(logits, y)
        self.log("train_loss", loss, prog_bar=True, on_epoch=True)  # this automatically logs `train_loss_epoch` at the end of epoch

        return loss

    def validation_step(self, batch, batch_idx):
        x, y = batch
        logits = self(x)

        loss = self.loss_fn(logits, y)
        self.log("val_loss", loss, prog_bar=True)

        preds = torch.argmax(logits, dim=1)
        acc = self.accuracy(preds, y)
        self.log("val_acc", acc, prog_bar=True)

        return loss
```

I'm not sure where the `train_acc_epoch` comes from as I am not calculating it anywhere, but I guess you can add it like this:
```python
acc = ...  # calculate here
self.log("train_acc_epoch", acc, on_step=False, on_epoch=True)
```

Here is what the plot look like:
![image](https://github.com/Lightning-AI/pytorch-lightning/assets/92390086/50efb585-82a0-4269-be66-2c781cf42f47)
